modules/core/src/lib/attribute-transition-utils.js

```diff
@@ -33,39 +82,41 @@ export function normalizeTransitionSettings(userSettings, layerSettings = {}) {
 // function, that looks like this: [A1, A2, A3, A4 (user `enter` fn), B1, B2, 0]. Note: the final
 // 0 in this buffer is because we never shrink buffers, only grow them, for performance reasons.
 export function padBuffer({
-  fromState,
-  toState,
+  buffer,
+  numInstances,
+  attribute,
   fromLength,
-  toLength,
   fromBufferLayout,
-  toBufferLayout,
-  size,
-  offset,
   getData = x => x
 }) {
+  // TODO: move the precisionMultiplier logic to the attribute when retrieving
+  // its `size` and `elementOffset`?
```

Comment:
`attribute.size` is used with `vertexAttribPointer` while the `size` here is used for manipulating buffer content on the CPU.

Reply:
I see - maybe a `getBufferContentLength` or something, since this calculation seems to be done a few times?

I'm not suggesting we do it as part of this work - just an idea.